src/undo/LineStyleUndoAction.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Xournal++
+ *
+ * Undo action for line style changes (Edit selection)
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <vector>
+
+#include "model/LineStyle.h"
+
+#include "UndoAction.h"
+
+class LineStyleUndoActionEntry;
+class Stroke;
+class Layer;
+class Redrawable;
+
+class LineStyleUndoAction: public UndoAction {
+public:
+    LineStyleUndoAction(const PageRef& page, Layer* layer);
+    virtual ~LineStyleUndoAction();
+
+public:
+    virtual bool undo(Control* control);
+    virtual bool redo(Control* control);
```

Comment:
Do not use virtual on inherited and overridden functions.
```suggestion
    bool undo(Control* control) override;
    bool redo(Control* control) override;
```